client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -336,14 +355,14 @@ public void processingFailure(Exception error) {
             state.failConnection(error);
         }
     }
-    
+
     /**
-     * @see SegmentOutputStream#write(java.nio.ByteBuffer,
-     *      java.util.concurrent.CompletableFuture)
+     * @see SegmentOutputStream#write(PendingEvent)
      */
     @Override
     @Synchronized
     public void write(PendingEvent event) throws SegmentSealedException {
+        checkState(!state.isAlreadySealed(), "Segment: {} is already sealed", segmentName);
```

Comment:
This is actually a problem. If a seal comes in over the wire concurrently to a write, the write will hold the lock in the caller and it will go first, but the exception will have already been set in the state so this will throw, before the write is added to the inflight so the caller will get the exception and drop the event, but it never made it into the inflight, so it won't be re-transmitted to the new segment.